app/src/main/java/com/exallium/mvvmcleanexample/app/binding/TextInputEditTextBindings.kt

```diff
@@ -3,16 +3,20 @@ package com.exallium.mvvmcleanexample.app.binding
 import android.databinding.BindingAdapter
 import android.support.design.widget.TextInputEditText
 import com.jakewharton.rxbinding2.widget.textChanges
-import io.reactivex.Observable
-import io.reactivex.functions.Consumer
+import io.reactivex.disposables.Disposable
 
 object TextInputEditTextBindings {
 
-    interface StringConsumer : Consumer<Observable<String>>
+    interface StringConsumer {
+        fun invoke(text: CharSequence)
+    }
 
     @JvmStatic
-    @BindingAdapter("bindTextChanges")
-    fun setChangesObservable(textInputEditText: TextInputEditText, consumer: StringConsumer) {
-        consumer.accept(textInputEditText.textChanges().map(CharSequence::toString))
+    @BindingAdapter("onTextChanged")
+    fun onTextChanged(textInputEditText: TextInputEditText, consumer: StringConsumer?) {
+        (textInputEditText.tag as? Disposable)?.dispose()
```

Comment:
I think this is a good change.

If the view model out-lives the view, say, due to a configuration change or something, we don't end up leaking the view.  I think.  If this is being done for such a change, that's cool.  I'm trying to think of ways which utilising tag might be unsafe.